### PR TITLE
CLI with install, uninstall and download version

### PR DESCRIPTION
**Pull request by pbochynski**
**Description**

Changes proposed in this pull request:

- install given version
- download charts from version
- uninstall Kyma
- provision/deprovision k3d cluster
- CLI with commands to install/upgrade Kyma, provision/deprovision k3d cluster
- flag to install new eventing (instead of knative), discovers: https://github.com/kyma-project/kyma/issues/10720
- upgrade test testing if connected application can survive the upgrade, discovers: https://github.com/kyma-project/kyma/issues/10759

### CODE REVIEW

File: tests/fast-integration/hack/cache-images.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  if [[ $1 =~ $regex ]]
+  then
+    registry=${BASH_REMATCH[1]//[$'\t\r\n']}
+    image=${BASH_REMATCH[2]//[$'\t\r\n']}
+    crane cp ${registry}${image} registry.localhost:5000/${image}
+  fi
+}
+
+docker exec -it k3d-kyma-server-0 sh -c "ctr images ls -q" >images.txt 
+cat images.txt | while read -r line; do cacheImage "$line"; done
```
Comment (crabtree) [inline tests/fast-integration/hack/cache-images.sh:12]:
```suggestion
images=`(docker exec -it k3d-kyma-server-0 sh -c "ctr images ls -q")`
while read -r line; do cacheImage "$line"; done < "${images}"
```

File: tests/fast-integration/installer/chart-downloader.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+const fs = require("fs");
+const { join } = require("path");
+var AdmZip = require("adm-zip");
```
Comment (crabtree) [inline tests/fast-integration/installer/chart-downloader.js:3]:
```suggestion
const AdmZip = require("adm-zip");
```

File: tests/fast-integration/installer/chart-downloader.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  debug("Resource path:",resourcesPath)
+  if (fs.existsSync(resourcesPath)) {
+    debug("The resources path already exists - download skipped")
+    return resourcesPath;
+  }
+  if (!fs.existsSync(dir)){
+    fs.mkdirSync(dir);
+  }
+
+  const zipFile = join(dir, branch + '.zip')
```
Comment (crabtree) [inline tests/fast-integration/installer/chart-downloader.js:43]:
```suggestion
  const zipFile = join(dir, `${branch}.zip`)
```

File: tests/fast-integration/installer/chart-downloader.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  const url = `https://codeload.github.com/${repo}/zip/${branch}`;
+  debug("Downloading Kyma charts from ",url);
+  await downloadFile(url, zipFile)
+  debug("Kyma charts downloaded");
+  var zip = new AdmZip(zipFile);
+  var zipEntries = zip.getEntries();
```
Comment (crabtree) [inline tests/fast-integration/installer/chart-downloader.js:49]:
```suggestion
  const zip = new AdmZip(zipFile);
  const zipEntries = zip.getEntries();
```

File: tests/fast-integration/package.json [hunk 5]
```diff
@@ -1,6 +1,14 @@
 {
-  "name": "fast-integration",
-  "version": "0.0.1",
+  "name": "kyma-js",
```
Comment (crabtree) [inline tests/fast-integration/package.json:2]: I'm a bit confused as we are mixing here (in this package) two things. The standalone kyma installation CLI and the tests that covers more thant the CLI responsibility.

Comment (pbochynski) [reply tests/fast-integration/package.json:2]: I was thinking about it. If we should separate the tool. But I treat it as a helper to run the integration tests locally. Long term kyma-cli should cover all the commands here. But it is not there yet. Keeping everything together helps a lot too speed up development. See the example: https://terminalizer.com/view/f55226d14700

File: tests/fast-integration/hack/cache-images.sh (hunk 1 above)
Comment (pbochynski) [reply tests/fast-integration/hack/cache-images.sh:12]: Hmm. It doesn't work for me. I had problems with multiline output. The temporary file solved my issues.

File: tests/fast-integration/installer/chart-downloader.js (hunk 2 above)
Comment (pbochynski) [reply tests/fast-integration/installer/chart-downloader.js]: Fixed

File: tests/fast-integration/installer/chart-downloader.js (hunk 4 above)
Comment (pbochynski) [reply tests/fast-integration/installer/chart-downloader.js]: Fixed

File: tests/fast-integration/installer/chart-downloader.js (hunk 3 above)
Comment (pbochynski) [reply tests/fast-integration/installer/chart-downloader.js]: Fixed

File: tests/fast-integration/hack/cache-images.sh (hunk 1 above)
Comment (pbochynski) [reply tests/fast-integration/hack/cache-images.sh]: I changed the file name to tmp-images.txt - it is now ignored.

File: tests/fast-integration/package.json (hunk 5 above)
Comment (pbochynski) [reply tests/fast-integration/package.json:2]: I added disclaimer to the overview in the README to point out that kyma-js is just a set of developers' tools, not the official CLI.

**Review by aerfio — CHANGES_REQUESTED** [review]

**Review by crabtree — APPROVED** [review]

File: tests/fast-integration/installer/chart-downloader.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+/**
+ * Downloads Kyma branch into temporary folder as zip, extracts charts, and returns full path to resources folder
+ * @param {Object} options Optional parameters
+ * @param {string} options.repo Github repository with Kyma charts, default: kyma-project/kyma
+ * @param {string} options.source Branch that should be downloaded from repository, default: master
+ * @return {string} Path to the resources folder
+ */
+async function downloadCharts(options) {
+  options = options || {};
+  const repo = options.repo || 'kyma-project/kyma'
```
[the remark this answers is not in the document]
Comment (pbochynski) [reply tests/fast-integration/installer/chart-downloader.js]: Done.

**Review by aerfio — APPROVED** [review]

**Review by klaudiagrz — APPROVED** [review]

**Pull request merged**